thorlcr/activities/keyedjoin/thkeyedjoinslave.cpp

```diff
@@ -2554,6 +2554,7 @@ class CKeyedJoinSlave : public CSlaveActivity, implements IJoinProcessor
                         waitingForDoneGroups = true;
                     }
                     waitingForDoneGroupsSem.wait();
+                    t.flushTimesIntermittantly();  // Need this even though time is spent blocked
```

Comment:
not sure I understand this comment..

As you say, this is blocked time really, so I think should be updating that here if anything.

Reply:
The local execute time is calculated from both the blocked time and the total cycles.  If we don't update the total cycles when the activity is blocked, then the execute time will not be calculated correctly.  In fact, we could have the situation of the totalCycles < blockedCycles, which doesn't make sense. (I assume the stats are updated intermittantly even if the activity is blocked or executing slowly?)